spec/02-integration/20-wasm/01-admin-api_spec.lua

```diff
@@ -387,4 +388,162 @@ describe("WASMX admin API [#" .. strategy .. "]", function()
+    })
+
+    assert(helpers.start_kong({
+      database = strategy,
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      wasm = "off",
+    }))
+
+    admin = helpers.admin_client()
+  end)
+
+  describe("/filter-chains", function()
+
+    describe("POST", function()
+      it("returns 400", function()
+        local res = admin:post("/filter-chains", {
+          headers = { ["Content-Type"] = "application/json" },
+          body = {
+            filters = { { name = "tests" } },
+            service = { id = service.id },
+          },
+        })
+
+        assert.response(res).has.status(400)
+      end)
+    end)
+
+    describe("GET", function()
+      it("returns 400", function()
+        local res = admin:get("/filter-chains")
+        assert.response(res).has.status(400)
+      end)
+    end)
+
+    describe("PATCH", function()
+      it("returns 400", function()
+        local res = admin:patch("/filter-chains/a-name", json {
+          tags = { "foo", "bar" },
+          enabled = false,
+          filters = {
+            { name = "tests", config = "123", enabled = true },
+            { name = "tests", config = "456", enabled = false },
+          },
+        })
+        assert.response(res).has.status(400)
+      end)
+    end)
+
+    describe("PUT", function()
+      it("returns 400", function()
+        local res = admin:put("/filter-chains/another-name", json {
+          tags = { "foo", "bar" },
+          enabled = false,
+          filters = {
+            { name = "tests", config = "123", enabled = true },
+            { name = "tests", config = "456", enabled = false },
+          },
+        })
+        assert.response(res).has.status(400)
+      end)
+    end)
+
+    describe("DELETE", function()
+      it("returns 400", function()
+        local res = admin:delete("/filter-chains/even-another-name")
+        assert.response(res).has.status(400)
+      end)
+    end)
+
+  end)
+
+  -- * /services/:service/filter-chains
+  -- * /services/:service/filter-chains/:chain
+  -- * /routes/:route/filter-chains
+  -- * /routes/:route/filter-chains/:chain
+  for _, rel in ipairs({ "service", "route" }) do
+
+  describe(fmt("/%ss/:%s/filter-chains", rel, rel), function()
+    local path, entity
+
+    before_each(function()
+      if rel == "service" then
+        entity = assert(bp.services:insert({}))
+      else
+        entity = assert(bp.routes:insert({ hosts = { "wasm.test" } }))
+      end
+
+      path = fmt("/%ss/%s/filter-chains", rel, entity.id)
+    end)
+
+    it("GET returns 400", function()
+      assert.response(
+        admin:get(path)
+      ).has.status(400)
+    end)
+
+    it("POST returns 400", function()
+      assert.response(
+        admin:post(path), {
+          headers = { ["Content-Type"] = "application/json" },
+          body = {
+            filters = { { name = "tests" } },
+            service = { id = service.id },
+          },
+        }
+      ).has.status(400)
```

Comment:
I think this is a typo--your request params are passed to `assert.response()` and not `admin:post()`:

```suggestion
        admin:post(path, json {
          filters = { { name = "tests" } },
          service = { id = service.id },
        }
      ).has.status(400)
```

Reply:
Yes, lol, parenthesis closed too soon 🤦